tests/suites/test_suite_opaque_test_driver.function

```diff
@@ -0,0 +1,643 @@
+                            opaque_test_driver_verify_hash(
+                                    &attr,
+                                    key,
+                                    OPAQUE_TEST_DRIVER_KEYHEADER_SIZE + 32,
+                                    PSA_ALG_ECDSA( PSA_ALG_SHA_256 ),
+                                    NULL,
+                                    16,
+                                    out_key,
+                                    out_key_len ) );
+    TEST_INVALID_PARAM_RET( PSA_ERROR_INVALID_ARGUMENT,
+                            opaque_test_driver_verify_hash(
+                                   &attr,
+                                   key,
+                                   OPAQUE_TEST_DRIVER_KEYHEADER_SIZE + 32,
+                                   PSA_ALG_ECDSA( PSA_ALG_SHA_256 ),
+                                   in_key,
+                                   16,
+                                   NULL,
+                                   out_key_len ) );
+
+exit:
+    psa_reset_key_attributes(&attr);
+    PSA_DONE( );
+    #undef OPQ_BUFSIZE
+}
+/* END_CASE */
+
+/* BEGIN_CASE */
+void opaque_test_driver_sign_verify( int key_type_arg, data_t *key_data,
+                                     int alg_arg, data_t *input_data )
+
+{
+    #define OPQ_BUFSIZE 128U
+    size_t key_bits;
+    uint8_t sign[OPQ_BUFSIZE];
+    size_t sign_size;
+    size_t sign_length = 0;
+    psa_key_attributes_t attr = PSA_KEY_ATTRIBUTES_INIT;
+
+    TEST_ASSERT( PSA_SUCCESS == psa_crypto_init() );
+
+    psa_set_key_usage_flags( &attr, PSA_KEY_USAGE_SIGN_HASH |
+                                    PSA_KEY_USAGE_VERIFY_HASH);
+    psa_set_key_algorithm( &attr, alg_arg );
+    psa_set_key_type( &attr, key_type_arg );
+    psa_set_key_bits( &attr, PSA_BYTES_TO_BITS( key_data->len -
+                                 OPAQUE_TEST_DRIVER_KEYHEADER_SIZE ) );
+
+    /* Sign the data with the given key. */
+    TEST_ASSERT( PSA_SUCCESS == opaque_test_driver_sign_hash(
+                                                         &attr,
+                                                         key_data->x,
+                                                         key_data->len,
+                                                         alg_arg,
+                                                         input_data->x,
+                                                         input_data->len,
+                                                         sign,
+                                                         OPQ_BUFSIZE,
+                                                         &sign_length ) );
+
+    key_bits = psa_get_key_bits( &attr );
+    sign_size = PSA_SIGN_OUTPUT_SIZE( key_type_arg, key_bits, alg_arg );
+
+    /* Check that the signature length looks sensible. */
+    TEST_ASSERT( sign_size != 0 );
+    TEST_ASSERT( sign_size <= PSA_SIGNATURE_MAX_SIZE );
+    TEST_ASSERT( sign_size <= OPQ_BUFSIZE );
```

Comment:
This is testing the `PSA_SIGN_OUTPUT_SIZE` macro, which is not an objective of this test suite.